Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx

```diff
@@ -0,0 +1,261 @@
+  }
+
+
+public:
+  //_________________________________________________________________________________________________
+  void init(framework::InitContext& ic)
+  {
+    Elec2Det = createElec2DetMapper<ElectronicMapperGenerated>();
+    fee2Solar = o2::mch::raw::createFeeLink2SolarMapper<ElectronicMapperGenerated>();
+    nrdhs = 0;
+
+    mPrint = ic.options().get<bool>("print");
+  }
+
+  //_________________________________________________________________________________________________
+  void run(framework::ProcessingContext& pc)
+  {
+    std::vector<o2::mch::Digit> digits;
+
+    // get the input buffer
+    auto& inputs = pc.inputs();
+    DPLRawParser parser(inputs, o2::framework::select("TF:MCH/RAWDATA"));
+
+    for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
+      // retrieving RDH v4
+      auto const* rdh = it.get_if<o2::header::RAWDataHeaderV4>();
+      // retrieving the raw pointer of the page
+      auto const* raw = it.raw();
+      // retrieving payload pointer of the page
+      //auto const* payload = it.data();
+      // size of payload
+      size_t payloadSize = it.size();
+      // offset of payload in the raw page
+      //size_t offset = it.offset();
+
+      if( payloadSize == 0 ) continue;
+
+
+      //std::cout<<"\n\npayloadSize: "<<payloadSize<<std::endl;
+      //std::cout<<"raw:     "<<(void*)raw<<std::endl;
+      //std::cout<<"payload: "<<(void*)payload<<std::endl;
+
+      gsl::span<const std::byte> buffer(reinterpret_cast<const std::byte*>(raw), sizeof(o2::header::RAWDataHeaderV4)+payloadSize);
+      decodeBuffer(buffer, digits);
+    }
+
+    for (auto&& input : pc.inputs()) {
+      //QcInfoLogger::GetInstance() << "run RawDataProcessor: input " << input.spec->binding << AliceO2::InfoLogger::InfoLogger::endm;
+
+      if (input.spec->binding != "readout")
+        continue;
+
+      const auto* header = o2::header::get<header::DataHeader*>(input.header);
+      if (!header)
+        continue;
+
+      //std::cout<<"payloadSize: "<<header->payloadSize<<std::endl;
+
+      gsl::span<const std::byte> buffer((const std::byte*)input.payload, header->payloadSize);
+      decodeBuffer(buffer, digits);
+    }
+
+    if(mPrint) {
+      for (auto d : digits) {
+        std::cout <<
+            " DE# " << d.getDetID() <<
+            " PadId " << d.getPadID() <<
+            " ADC " << d.getADC() <<
+            " time "<< d.getTimeStamp() <<
+            std::endl;
+      }
+    }
+
+    const size_t OUT_SIZE = sizeof(o2::mch::Digit) * digits.size();
+
+    /// send the output buffer via DPL
+    char* outbuffer = NULL;
+    outbuffer = (char*)realloc(outbuffer, OUT_SIZE);
+    memcpy(outbuffer, digits.data(), OUT_SIZE);
+
+    // create the output message
+    auto freefct = [](void* data, void*) { free(data); };
+    pc.outputs().adoptChunk(Output{ "MCH", "DIGITS", 0 }, outbuffer, OUT_SIZE, freefct, nullptr);
+  }
+
+private:
+  std::function<std::optional<DsDetId>(DsElecId)> Elec2Det;
```

Comment:
class members variable names should start with m (see https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html#Variable_Names)

Reply:
Ok. I also removed the un-needed fee2Solar member.